<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Human Protein Atlas in R &bull; hpar</title><!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous"><script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous"><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">hpar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Human Protein Atlas in R</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      Computational Proteomics Unit, Cambridge, UK<br></address></div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The Human Protein Atlas (HPA) is a systematic study oh the human proteome using antibody-based proteomics. Multiple tissues and cell lines are systematically assayed affinity-purified antibodies and confocal microscopy. The <em>hpar</em> package is an R interface to the HPA project. It distributes three data sets, provides functionality to query these and to access detailed information pages, including confocal microscopy images available on the HPA web page.</p>
    </div>
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#introduction" class="anchor"> </a></body></html>Introduction</h1>
<div id="the-hpa-project" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#the-hpa-project" class="anchor"> </a></body></html>The HPA project</h2>
<p>From the <a href="http://www.proteinatlas.org/">Human Protein Atlas</a> <span class="citation">(Uhl&eacute;n et al. 2005; Uhlen et al. 2010)</span> site:</p>
<blockquote>
<p>The Swedish Human Protein Atlas project, funded by the Knut and Alice Wallenberg Foundation, has been set up to allow for a systematic exploration of the human proteome using Antibody-Based Proteomics. This is accomplished by combining high-throughput generation of affinity-purified antibodies with protein profiling in a multitude of tissues and cells assembled in tissue microarrays. Confocal microscopy analysis using human cell lines is performed for more detailed protein localisation. The program hosts the Human Protein Atlas portal with expression profiles of human proteins in tissues and cells.</p>
</blockquote>
<p>The <em><a href="http://bioconductor.org/packages/hpar">hpar</a></em> package provides access to HPA data from the R interface. It also distributes the following data sets:</p>
<ul><li><p><code>hpaNormalTissue</code> <em>Normal tissue data</em>: Expression profiles for proteins in human tissues based on immunohistochemisty using tissue micro arrays. The comma-separated file includes Ensembl gene identifier (&ldquo;Gene&rdquo;), tissue name (&ldquo;Tissue&rdquo;), annotated cell type (&ldquo;Cell type&rdquo;), expression value (&ldquo;Level&rdquo;), the type of annotation (annotated protein expression (APE), based on more than one antibody, or staining, based on one antibody only) (&ldquo;Expression type&rdquo;), and the reliability or validation of the expression value (&ldquo;Reliability&rdquo;).}</p></li>
<li><p><code>hpaCancer</code> <em>Cancer tumor data</em>: Staining profiles for proteins in human tumor tissue based on immunohistochemisty using tissue micro arrays. The comma-separated file includes Ensembl gene identifier (&ldquo;Gene&rdquo;), tumor name (&ldquo;Tumor&rdquo;), staining value (&ldquo;Level&rdquo;), the number of patients that stain for this staining value (&ldquo;Count patients&rdquo;), the total amount of patients for this tumor type (&ldquo;Total patients&rdquo;) and the type of annotation staining (&ldquo;Expression type&rdquo;). }</p></li>
<li><p><code>rnaGeneTissue</code> <em>RNA gene data</em>: RNA levels in 45 cell lines and 32 tissues based on RNA-seq. The comma-separated file includes Ensembl gene identifier (&ldquo;Gene&rdquo;), analysed sample (&ldquo;Sample&rdquo;), fragments per kilobase of transcript per million fragments mapped (&ldquo;Value&rdquo; and &ldquo;Unit&rdquo;), and abundance class (&ldquo;Abundance&rdquo;). }</p></li>
<li><p><code>rnaGeneCellLine</code> <em>RNA gene data</em>: RNA levels in 45 cell lines and 32 tissues based on RNA-seq. The comma-separated file includes Ensembl gene identifier (&ldquo;Gene&rdquo;), analysed sample (&ldquo;Sample&rdquo;), fragments per kilobase of transcript per million fragments mapped (&ldquo;Value&rdquo; and &ldquo;Unit&rdquo;), and abundance class (&ldquo;Abundance&rdquo;). }</p></li>
<li><p><code>hpaSubcellularLoc</code> <em>Subcellular location data</em>: Subcellular localization of proteins based on immunofluorescently stained cells. The comma-separated file includes Ensembl gene identifier (&ldquo;Gene&rdquo;), main subcellular location of the protein (&ldquo;Main location&rdquo;), other locations (&ldquo;Other location&rdquo;), the type of annotation (annotated protein expression (APE), based on more than one antibody, or staining, based on one antibody only) (&ldquo;Expression type&rdquo;), and the reliability or validation of the expression value (&ldquo;Reliability&rdquo;). }</p></li>
<li><p><code>hpaSubcellularLoc14</code>: Same as above, for version 14.</p></li>
</ul></div>
<div id="hpa-data-usage-policy" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#hpa-data-usage-policy" class="anchor"> </a></body></html>HPA data usage policy</h2>
<p>The use of data and images from the HPA in publications and presentations is permitted provided that the following conditions are met:</p>
<ul><li>The publication and/or presentation are solely for informational and non-commercial purposes.</li>
<li>The source of the data and/or image is referred to the HPA site<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a> and/or one or more of our publications are cited.</li>
</ul></div>
<div id="installation" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#installation" class="anchor"> </a></body></html>Installation</h2>
<p><em><a href="http://bioconductor.org/packages/hpar">hpar</a></em> is available through the Bioconductor project. Details about the package and the installation procedure can be found on its <a href="http://bioconductor.org/packages/devel/bioc/html/hpar.html">landing page</a>. To install using the dedicated Bioconductor infrastructure, run :</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">"http://bioconductor.org/biocLite.R"</span>)  
## or, if you have already used the above before 
<span class="kw">library</span>(<span class="st">"BiocInstaller"</span>) ## and to install the package 
<span class="kw">biocLite</span>(<span class="st">"hpar"</span>)</code></pre>
<p>After installation, <em><a href="http://bioconductor.org/packages/hpar">hpar</a></em> will have to be explicitly loaded with</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"hpar"</span>)</code></pre>
<p>so that all the package&rsquo;s functionality and data is available to the user.</p>
</div>
</div>
<div id="the-hpar-package" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#the-hpar-package" class="anchor"> </a></body></html>The <em><a href="http://bioconductor.org/packages/hpar">hpar</a></em> package</h1>
<div id="data-sets" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#data-sets" class="anchor"> </a></body></html>Data sets</h2>
<p>The data sets described above can be loaded with the <code>data</code> function, as illustrated below for <code>hpaNormalTissue</code> below. Each data set is a <code>data.frame</code> and can be easily manipulated using standard R functionality. The code chunk below illustrates some of its properties.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(hpaNormalTissue)
<span class="kw">dim</span>(hpaNormalTissue)</code></pre>
<pre><code>## [1] 1159341       7</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(hpaNormalTissue)</code></pre>
<pre><code>## [1] "Gene"            "Gene.name"       "Tissue"          "Cell.type"      
## [5] "Level"           "Expression.type" "Reliability"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">## Number of genes
<span class="kw">length</span>(<span class="kw">unique</span>(hpaNormalTissue$Gene))</code></pre>
<pre><code>## [1] 14578</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">## Number of cell types
<span class="kw">length</span>(<span class="kw">unique</span>(hpaNormalTissue$Cell.type))</code></pre>
<pre><code>## [1] 44</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">levels</span>(hpaNormalTissue$Cell.type))</code></pre>
<pre><code>## [1] "adipocytes"                  "bile duct cells"            
## [3] "cells in endometrial stroma" "cells in glomeruli"         
## [5] "cells in granular layer"     "cells in molecular layer"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">## Number of tissues
<span class="kw">length</span>(<span class="kw">unique</span>(hpaNormalTissue$Tissue))</code></pre>
<pre><code>## [1] 48</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">levels</span>(hpaNormalTissue$Tissue))</code></pre>
<pre><code>## [1] "adrenal gland" "appendix"      "bone marrow"   "breast"       
## [5] "bronchus"      "cerebellum"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(hpaNormalTissue$Expression.type)</code></pre>
<pre><code>## 
##      APE Staining 
##   752442   406899</code></pre>
</div>
<div id="hpa-interface" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#hpa-interface" class="anchor"> </a></body></html>HPA interface</h2>
<p>The package provides a interface to the HPA data. The <code>getHpa</code> allows to query the data sets described above. It takes three arguments, <code>id</code>, <code>hpadata</code> and <code>type</code>, that control the query, what data set to interrogate and how to report results respectively. The HPA data uses Ensembl gene identifiers and <code>id</code> must be a valid identifier. <code>hpadata</code> must be one of available dataset. <code>type</code> can be either <code>"data"</code> or <code>"details"</code>. The former is the default and returns a <code>data.frame</code> containing the information relevant to <code>id</code>. It is also possible to obtained detailed information, (including cell images) as web pages, directly from the HPA web page, using <code>"details"</code>.</p>
<p>We will illustrate this functionality with using the TSPAN6 (tetraspanin 6) gene (ENSG00000000003) as example.</p>
<pre class="sourceCode r"><code class="sourceCode r">id &lt;-<span class="st"> "ENSG00000000003"</span>
<span class="kw">head</span>(<span class="kw"><a href="../reference/getHpa.html">getHpa</a></span>(id, <span class="dt">hpadata =</span> <span class="st">"hpaNormalTissue"</span>))</code></pre>
<pre><code>##              Gene Gene.name        Tissue           Cell.type        Level
## 1 ENSG00000000003    TSPAN6 adrenal gland     glandular cells Not detected
## 2 ENSG00000000003    TSPAN6      appendix     glandular cells       Medium
## 3 ENSG00000000003    TSPAN6      appendix     lymphoid tissue Not detected
## 4 ENSG00000000003    TSPAN6   bone marrow hematopoietic cells Not detected
## 5 ENSG00000000003    TSPAN6        breast          adipocytes Not detected
## 6 ENSG00000000003    TSPAN6        breast     glandular cells         High
##   Expression.type Reliability
## 1             APE   Uncertain
## 2             APE   Uncertain
## 3             APE   Uncertain
## 4             APE   Uncertain
## 5             APE   Uncertain
## 6             APE   Uncertain</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getHpa.html">getHpa</a></span>(id, <span class="dt">hpadata =</span> <span class="st">"hpaSubcellularLoc"</span>)</code></pre>
<pre><code>##              Gene Gene.name Main.location Other.location Expression.type
## 1 ENSG00000000003    TSPAN6     Cytoplasm                            APE
##   Reliability Main.location.GO.id Other.location.GO.id
## 1   Uncertain          GO:0005737</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="../reference/getHpa.html">getHpa</a></span>(id, <span class="dt">hpadata =</span> <span class="st">"rnaGeneCellLine"</span>))</code></pre>
<pre><code>##              Gene Gene.name  Sample Value Unit Abundance
## 1 ENSG00000000003    TSPAN6   A-431  17.5 FPKM       Low
## 2 ENSG00000000003    TSPAN6    A549  27.2 FPKM    Medium
## 3 ENSG00000000003    TSPAN6  AN3-CA  32.9 FPKM    Medium
## 4 ENSG00000000003    TSPAN6    BEWO  26.0 FPKM    Medium
## 5 ENSG00000000003    TSPAN6  CACO-2  56.2 FPKM      High
## 6 ENSG00000000003    TSPAN6 CAPAN-2  30.0 FPKM    Medium</code></pre>
<p>If we ask for <code>"detail"</code>, a browser page pointing to the relevant page is open (see figure below)</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getHpa.html">getHpa</a></span>(id, <span class="dt">type =</span> <span class="st">"details"</span>)</code></pre>
<div class="figure">
<img src="hpa.png" alt="The HPA web page for the tetraspanin 6 gene (ENSG00000000003)."><p class="caption">The HPA web page for the tetraspanin 6 gene (ENSG00000000003).</p>
</div>
<p>If a user is interested specifically in one data set, it is possible to set <code>hpadata</code> globally and omit it in <code>getHpa</code>. This is done by setting the <code>hpar</code> options <code>hpardata</code> with the <code>setHparOptions</code> function. The current default data set can be tested with <code>getHparOptions</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getHparOptions.html">getHparOptions</a></span>()</code></pre>
<pre><code>## $hpar
## $hpar$hpadata
## [1] "hpaNormalTissue"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/setHparOptions.html">setHparOptions</a></span>(<span class="dt">hpadata =</span> <span class="st">"hpaSubcellularLoc"</span>)
<span class="kw"><a href="../reference/getHparOptions.html">getHparOptions</a></span>()</code></pre>
<pre><code>## $hpar
## $hpar$hpadata
## [1] "hpaSubcellularLoc"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getHpa.html">getHpa</a></span>(id)</code></pre>
<pre><code>##              Gene Gene.name Main.location Other.location Expression.type
## 1 ENSG00000000003    TSPAN6     Cytoplasm                            APE
##   Reliability Main.location.GO.id Other.location.GO.id
## 1   Uncertain          GO:0005737</code></pre>
</div>
<div id="hpa-release-information" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#hpa-release-information" class="anchor"> </a></body></html>HPA release information</h2>
<p>Information about the HPA release used to build the installed</p>
<p><em><a href="http://bioconductor.org/packages/hpar">hpar</a></em> package can be accessed with <code>getHpaVersion</code>, <code>getHpaDate</code> and <code>getHpaEnsembl</code>. Full release details can be found on the <a href="http://www.proteinatlas.org/about/releases">HPA release history</a> page.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getHpaDate.html">getHpaVersion</a></span>()</code></pre>
<pre><code>## version 
##    "15"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getHpaDate.html">getHpaDate</a></span>()</code></pre>
<pre><code>##         date 
## "2016.04.11"</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getHpaDate.html">getHpaEnsembl</a></span>()</code></pre>
<pre><code>## ensembl 
## "78.38"</code></pre>
</div>
</div>
<div id="a-small-use-case" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#a-small-use-case" class="anchor"> </a></body></html>A small use case</h1>
<p>Let&rsquo;s compare the subcellular localisation annotation obtained from the HPA subcellular location data set and the information available in the Bioconductor annotation packages.</p>
<pre class="sourceCode r"><code class="sourceCode r">id &lt;-<span class="st"> "ENSG00000001460"</span>
<span class="kw"><a href="../reference/getHpa.html">getHpa</a></span>(id, <span class="st">"hpaSubcellularLoc"</span>)</code></pre>
<pre><code>##              Gene Gene.name Main.location   Other.location Expression.type
## 8 ENSG00000001460     STPG1       Nucleus Nuclear membrane             APE
##   Reliability Main.location.GO.id Other.location.GO.id
## 8  Supportive          GO:0005634           GO:0031965</code></pre>
<p>Below, we first extract all cellular component GO terms available for <code>id</code> from the <em><a href="http://bioconductor.org/packages/org.Hs.eg.db">org.Hs.eg.db</a></em> human annotation and then retrieve their term definitions using the <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> database.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"org.Hs.eg.db"</span>)
<span class="kw">library</span>(<span class="st">"GO.db"</span>)
ans &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db, <span class="dt">keys =</span> id, 
              <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">"ENSEMBL"</span>, <span class="st">"GO"</span>, <span class="st">"ONTOLOGY"</span>), 
              <span class="dt">keytype =</span> <span class="st">"ENSEMBL"</span>)</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">ans &lt;-<span class="st"> </span>ans[ans$ONTOLOGY ==<span class="st"> "CC"</span>, ]
ans</code></pre>
<pre><code>##           ENSEMBL         GO EVIDENCE ONTOLOGY
## 1 ENSG00000001460 GO:0005634      IEA       CC
## 2 ENSG00000001460 GO:0005737      IEA       CC</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sapply</span>(<span class="kw">as.list</span>(GOTERM[ans$GO]), slot, <span class="st">"Term"</span>)</code></pre>
<pre><code>##  GO:0005634  GO:0005737 
##   "nucleus" "cytoplasm"</code></pre>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#session-information" class="anchor"> </a></body></html>Session information}</h1>
<pre><code>## R Under development (unstable) (2016-12-09 r71771)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] GO.db_3.4.0          org.Hs.eg.db_3.4.0   AnnotationDbi_1.37.0
##  [4] IRanges_2.9.14       S4Vectors_0.13.5     Biobase_2.35.0      
##  [7] BiocGenerics_0.21.1  BiocStyle_2.3.28     hpar_1.17.1         
## [10] pkgdown_0.1.0.9000  
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.8.2        msdata_0.15.0        compiler_3.4.0      
##  [4] tools_3.4.0          highlight_0.4.7      testthat_1.0.2      
##  [7] digest_0.6.10        pkgbuild_0.0.0.9000  pkgload_0.0.0.9000  
## [10] evaluate_0.10        memoise_1.0.0        RSQLite_1.1-1       
## [13] tibble_1.2           DBI_0.5-1            yaml_2.1.14         
## [16] withr_1.0.2          stringr_1.1.0        roxygen2_5.0.1      
## [19] xml2_1.0.0.9002      knitr_1.15.1         desc_1.0.1          
## [22] devtools_1.12.0.9000 rprojroot_1.1        R6_2.2.0            
## [25] rmarkdown_1.3        purrr_0.2.2          magrittr_1.5        
## [28] whisker_0.3-2        backports_1.0.4      htmltools_0.3.5     
## [31] assertthat_0.1       stringi_1.1.2        lazyeval_0.2.0      
## [34] crayon_1.3.2</code></pre>
<div class="references">
<p>Uhlen, Mathias, Per Oksvold, Linn Fagerberg, Emma Lundberg, Kalle Jonasson, Mattias Forsberg, Martin Zwahlen, et al. 2010. &ldquo;Towards a Knowledge-Based Human Protein Atlas.&rdquo; <em>Nature Biotechnology</em> 28 (12). Nature Publishing Group, a division of Macmillan Publishers Limited. All Rights Reserved.: 1248&ndash;50. doi:<a href="http://dx.doi.org/10.1038/nbt1210-1248">10.1038/nbt1210-1248</a>. <a href="http://dx.doi.org/10.1038/nbt1210-1248">http://dx.doi.org/10.1038/nbt1210-1248</a>.</p>
<p>Uhl&eacute;n, Mathias, Erik Bj&ouml;rling, Charlotta Agaton, Cristina Al-Khalili A. Szigyarto, Bahram Amini, Elisabet Andersen, Ann-Catrin C. Andersson, et al. 2005. &ldquo;A Human Protein Atlas for Normal and Cancer Tissues Based on Antibody Proteomics.&rdquo; <em>Molecular &amp; Cellular Proteomics : MCP</em> 4 (12). Department of Biotechnology, AlbaNova University Center, Royal Institute of Technology (KTH), SE-106 91 Stockholm, Sweden. mathias.uhlen@biotech.kth.se: 1920&ndash;32. doi:<a href="http://dx.doi.org/10.1074/mcp.M500279-MCP200">10.1074/mcp.M500279-MCP200</a>. <a href="http://dx.doi.org/10.1074/mcp.M500279-MCP200">http://dx.doi.org/10.1074/mcp.M500279-MCP200</a>.</p>
</div>
</div>
<div class="footnotes">
<hr><ol><li id="fn1"><p>www.proteinatlas.org<a href="#fnref1">&#8617;</a></p></li>
</ol></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked"><li><a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked"><li><a href="#the-hpa-project">The HPA project</a></li>
      <li><a href="#hpa-data-usage-policy">HPA data usage policy</a></li>
      <li><a href="#installation">Installation</a></li>
      </ul></li>
      <li><a href="#the-hpar-package">The <em><a href="http://bioconductor.org/packages/hpar">hpar</a></em> package</a><ul class="nav nav-pills nav-stacked"><li><a href="#data-sets">Data sets</a></li>
      <li><a href="#hpa-interface">HPA interface</a></li>
      <li><a href="#hpa-release-information">HPA release information</a></li>
      </ul></li>
      <li><a href="#a-small-use-case">A small use case</a></li>
      <li><a href="#session-information">Session information}</a></li>
      </ul></div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer></div>

  </body></html>
