<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Human Protein Atlas in R — hpar • hpar</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Human Protein Atlas in R — hpar" />

<meta property="og:description" content="This package provides a simple interface to the Human Protein
  Atlas. From the Human Protein Atlas Project page:
  The Swedish Human Protein Atlas project, funded by the Knut and
  Alice Wallenberg Foundation, has been set up to allow for a systematic
  exploration of the human proteome using Antibody-Based
  Proteomics. This is accomplished by combining high-throughput
  generation of affinity-purified antibodies with protein profiling in a
  multitude of tissues and cells assembled in tissue
  microarrays. Confocal microscopy analysis using human cell lines is
  performed for more detailed protein localization. The program hosts
  the Human Protein Atlas portal with expression profiles of human
  proteins in tissues and cells." />
<meta name="twitter:card" content="summary" />


<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hpar</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.23.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/hpar.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Human Protein Atlas in R</h1>
    
    <div class="hidden name"><code>hpar.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This package provides a simple interface to the Human Protein
  Atlas. From the Human Protein Atlas Project page:
  The Swedish Human Protein Atlas project, funded by the Knut and
  Alice Wallenberg Foundation, has been set up to allow for a systematic
  exploration of the human proteome using Antibody-Based
  Proteomics. This is accomplished by combining high-throughput
  generation of affinity-purified antibodies with protein profiling in a
  multitude of tissues and cells assembled in tissue
  microarrays. Confocal microscopy analysis using human cell lines is
  performed for more detailed protein localization. The program hosts
  the Human Protein Atlas portal with expression profiles of human
  proteins in tissues and cells.</p>
    
    </div>

        
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Several flat files are distributed by the HPA project and
  available within the package as <code>data.frame</code>s. The description
  below is taken from the HPA site:</p>
<dl class='dl-horizontal'>
    <dt>hpaNormalTissue</dt><dd><p><b>Normal tissue data</b>: Expression profiles
    for proteins in human tissues based on immunohistochemisty using
    tissue micro arrays. The comma-separated file includes Ensembl gene
    identifier ("Gene"), tissue name ("Tissue"), annotated cell type
    ("Cell type"), expression value ("Level"), the type of annotation
    (annotated protein expression (APE), based on more than one antibody,
    or staining, based on one antibody only) ("Expression type"), and the
    reliability or validation of the expression value ("Reliability").</p></dd>
    <dt>hpaCancer</dt><dd><p><b>Cancer tumor data</b>: Staining profiles for
    proteins in human tumor tissue based on immunohistochemisty using
    tissue micro arrays. The comma-separated file includes Ensembl gene
    identifier ("Gene"), tumor name ("Tumor"), staining value ("Level"),
    the number of patients that stain for this staining value
    ("Count patients"), the total amount of patients for this tumor type
    ("Total patients") and the type of annotation staining
    ("Expression type").</p></dd>
    <dt>rnaGeneTissue</dt><dd><p><b>RNA gene data</b>: RNA levels in 45 cell lines
    and 32 tissues based on RNA-seq. The comma-separated file includes
    Ensembl gene identifier ("Gene"), analysed sample ("Sample"),
    fragments per kilobase of transcript per million fragments mapped
    ("Value" and "Unit"), and abundance class ("Abundance").</p></dd>
    <dt>rnaGeneCellLine</dt><dd><p><b>RNA gene data</b>: RNA levels in 45 cell
    lines and 32 tissues based on RNA-seq. The comma-separated file
    includes Ensembl gene identifier ("Gene"), analysed sample
    ("Sample"), fragments per kilobase of transcript per million
    fragments mapped ("Value" and "Unit"), and abundance class
    ("Abundance").</p></dd>
    <dt>hpaSubcellularLoc</dt><dd><p><b>Subcellular location data</b>:
    Subcellular localization of proteins based on
    immunofluorescently stained cells. The comma-separated file
    includes Ensembl gene identifier ("Gene"), main subcellular
    location of the protein ("Main location"), other locations
    ("Other location"), the type of annotation (annotated protein
    expression (APE), based on more than one antibody, or staining,
    based on one antibody only) ("Expression type"), and the
    reliability or validation of the expression value
    ("Reliability").</p></dd>
    <dt>hpaSubcellularLoc14</dt><dd><p>Same as above, for version 14.</p></dd>
    </dl>
    <p>Detailed description for gene entries and images and not included in
the package but can be accessed from within the <span style="R">R</span> environment through
a web browser while on-line.</p>  
<p>The full data sets can be individually loaded using the
<code>data</code> function (see example below).
Data about individual genes of interest can retrived with the
<code><a href='getHpa.html'>getHpa</a></code> function.</p>
<p>HPA data usage policy: The use of data and images from this site in
publications and presentations is permitted provided that the
following conditions are met:</p>  
<ol>
<li><p>The publication and/or presentation are solely for
  informational and non-commercial purposes.</p></li>
<li><p>The source of the data and/or image is referred to this site
  (www.proteinatlas.org) and/or one or more of our publications are
  cited.</p></li>
</ol>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='getHpaDate.html'>getHpaDate</a></code> for release information. Gene-specific
  information should be accessed using the <code><a href='getHpa.html'>getHpa</a></code>
  function.</p>
<p>The package vignette can be accessed with <code><a href='../articles/hpar.html'>vignette("hpar")</a></code>.</p></div>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
    <p>See the Human Protein Atlas Project page
  <a href='http://www.proteinatlas.org/'>http://www.proteinatlas.org/</a> and
  <a href='http://www.proteinatlas.org/about/download'>http://www.proteinatlas.org/about/download</a> for more details and
  documentation.</p>
    <p>Uhlen et al (2010). Towards a knowledge-based Human Protein Atlas. Nat
  Biotechnol. 28(12):1248-50.</p>  
<p>Berglund et al (2008). A gene-centric Human Protein Atlas for
  expression profiles based on antibodies. Mol Cell
  Proteomics. 7(10):2019-27.</p>  
<p>Uhlen et al (2005). A human protein atlas for normal and cancer
  tissues based on antibody proteomics. Mol Cell
  Proteomics. 4(12):1920-1932.</p>  
<p>Ponten et al (2008). The Human Protein Atlas - a tool for pathology. J
  Pathology. 216(4):387-93.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'>allHparData</span>()</div><div class='output co'>#&gt; [1] "hpaNormalTissue"     "rnaGeneCellLine"     "rnaGeneTissue"      
#&gt; [4] "hpaSubcellularLoc"   "hpaSubcellularLoc14" "hpaCancer"          </div><div class='input'>
<span class='fu'>data</span>(<span class='no'>hpaNormalTissue</span>)
<span class='fu'>head</span>(<span class='no'>hpaNormalTissue</span>)</div><div class='output co'>#&gt;              Gene Gene.name        Tissue           Cell.type        Level
#&gt; 1 ENSG00000000003    TSPAN6 adrenal gland     glandular cells Not detected
#&gt; 2 ENSG00000000003    TSPAN6      appendix     glandular cells       Medium
#&gt; 3 ENSG00000000003    TSPAN6      appendix     lymphoid tissue Not detected
#&gt; 4 ENSG00000000003    TSPAN6   bone marrow hematopoietic cells Not detected
#&gt; 5 ENSG00000000003    TSPAN6        breast          adipocytes Not detected
#&gt; 6 ENSG00000000003    TSPAN6        breast     glandular cells         High
#&gt;   Reliability
#&gt; 1    Approved
#&gt; 2    Approved
#&gt; 3    Approved
#&gt; 4    Approved
#&gt; 5    Approved
#&gt; 6    Approved</div><div class='input'>
<span class='fu'>data</span>(<span class='no'>rnaGeneCellLine</span>)
<span class='fu'>head</span>(<span class='no'>rnaGeneCellLine</span>)</div><div class='output co'>#&gt;              Gene Gene.name    Sample Value Unit
#&gt; 1 ENSG00000000003    TSPAN6     A-431  27.8  TPM
#&gt; 2 ENSG00000000003    TSPAN6      A549  37.6  TPM
#&gt; 3 ENSG00000000003    TSPAN6      AF22 108.1  TPM
#&gt; 4 ENSG00000000003    TSPAN6    AN3-CA  51.8  TPM
#&gt; 5 ENSG00000000003    TSPAN6  ASC diff  32.3  TPM
#&gt; 6 ENSG00000000003    TSPAN6 ASC TERT1  17.7  TPM</div><div class='input'>
<span class='fu'>data</span>(<span class='no'>rnaGeneTissue</span>)
<span class='fu'>head</span>(<span class='no'>rnaGeneTissue</span>)</div><div class='output co'>#&gt;              Gene Gene.name          Sample Value Unit
#&gt; 1 ENSG00000000003    TSPAN6  adipose tissue  31.5  TPM
#&gt; 2 ENSG00000000003    TSPAN6   adrenal gland  26.4  TPM
#&gt; 3 ENSG00000000003    TSPAN6        appendix   9.2  TPM
#&gt; 4 ENSG00000000003    TSPAN6     bone marrow   0.7  TPM
#&gt; 5 ENSG00000000003    TSPAN6          breast  53.4  TPM
#&gt; 6 ENSG00000000003    TSPAN6 cerebral cortex  18.5  TPM</div><div class='input'>
<span class='fu'>data</span>(<span class='no'>hpaSubcellularLoc</span>)
<span class='fu'>head</span>(<span class='no'>hpaSubcellularLoc</span>)</div><div class='output co'>#&gt;              Gene Gene.name Reliability Enhanced Supported
#&gt; 1 ENSG00000000003    TSPAN6    Approved                   
#&gt; 2 ENSG00000000457     SCYL3   Uncertain                   
#&gt; 3 ENSG00000000460  C1orf112    Approved                   
#&gt; 4 ENSG00000000938       FGR    Approved                   
#&gt; 5 ENSG00000000971       CFH    Approved                   
#&gt; 6 ENSG00000001084      GCLC    Approved                   
#&gt;                    Approved                   Uncertain
#&gt; 1                   Cytosol                            
#&gt; 2                           Microtubules;Nuclear bodies
#&gt; 3              Mitochondria                            
#&gt; 4 Aggresome;Plasma membrane                            
#&gt; 5                  Vesicles                            
#&gt; 6  Cytosol;Nucleoli;Nucleus                            
#&gt;   Single.cell.variation.intensity Single.cell.variation.spatial
#&gt; 1                                                              
#&gt; 2                                                              
#&gt; 3                                                              
#&gt; 4                                                              
#&gt; 5                                                              
#&gt; 6                                                              
#&gt;   Cell.cycle.dependency
#&gt; 1                      
#&gt; 2                      
#&gt; 3                      
#&gt; 4                      
#&gt; 5                      
#&gt; 6                      
#&gt;                                                             GO.id
#&gt; 1                                            Cytosol (GO:0005829)
#&gt; 2           Microtubules (GO:0015630);Nuclear bodies (GO:0016604)
#&gt; 3                                       Mitochondria (GO:0005739)
#&gt; 4             Aggresome (GO:0016235);Plasma membrane (GO:0005886)
#&gt; 5                                           Vesicles (GO:0043231)
#&gt; 6 Cytosol (GO:0005829);Nucleoli (GO:0005730);Nucleus (GO:0005634)</div><div class='input'>
<span class='fu'>data</span>(<span class='no'>hpaSubcellularLoc14</span>)
<span class='fu'>head</span>(<span class='no'>hpaSubcellularLoc14</span>)</div><div class='output co'>#&gt;              Gene                         Main.location
#&gt; 1 ENSG00000000003                             Cytoplasm
#&gt; 2 ENSG00000000457           Cytoskeleton (Microtubules)
#&gt; 3 ENSG00000000460 Nucleus but not nucleoli;Mitochondria
#&gt; 4 ENSG00000001036 Nucleus but not nucleoli;Mitochondria
#&gt; 5 ENSG00000001084                      Nucleus;Nucleoli
#&gt; 6 ENSG00000001460                               Nucleus
#&gt;                             Other.location Expression.type Reliability
#&gt; 1                                                      APE   Uncertain
#&gt; 2 Nucleus but not nucleoli;Golgi apparatus             APE   Uncertain
#&gt; 3                                                      APE   Uncertain
#&gt; 4                                                      APE   Uncertain
#&gt; 5                                Cytoplasm             APE  Supportive
#&gt; 6                         Nuclear membrane             APE  Supportive</div><div class='input'>
<span class='fu'>data</span>(<span class='no'>hpaCancer</span>)
<span class='fu'>head</span>(<span class='no'>hpaCancer</span>)</div><div class='output co'>#&gt;              Gene Gene.name             Cancer High Medium Low Not.detected
#&gt; 1 ENSG00000000003    TSPAN6      breast cancer    1      7   2            2
#&gt; 2 ENSG00000000003    TSPAN6          carcinoid    0      1   1            2
#&gt; 3 ENSG00000000003    TSPAN6    cervical cancer   11      1   0            0
#&gt; 4 ENSG00000000003    TSPAN6  colorectal cancer    0      6   2            2
#&gt; 5 ENSG00000000003    TSPAN6 endometrial cancer   10      2   0            0
#&gt; 6 ENSG00000000003    TSPAN6             glioma    0      0   0           11
#&gt;   prognostic.favourable unprognostic.favourable prognostic.unfavourable
#&gt; 1                    NA                0.077120                      NA
#&gt; 2                    NA                      NA                      NA
#&gt; 3                    NA                0.089670                      NA
#&gt; 4                    NA                0.035620                      NA
#&gt; 5                    NA                      NA                      NA
#&gt; 6                    NA                0.003617                      NA
#&gt;   unprognostic.unfavourable
#&gt; 1                        NA
#&gt; 2                        NA
#&gt; 3                        NA
#&gt; 4                        NA
#&gt; 5                    0.2567
#&gt; 6                        NA</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#details">Details</a></li>

      <li><a href="#see-also">See also</a></li>

      <li><a href="#references">References</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
  Laurent Gatto &lt;lg390@cam.ac.uk&gt;

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

